### DISCUSSION TITLE
Welcome to Winget-Install Discussions!

### DISCUSSION THREAD

**Discussion by Romanitho** [Romanitho/Winget-Install#16] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by KnifMelti** [comment]
Hola!
I'm a dinosaur (59 years old), working as an application specialist/packager/developer/problem solver/etc., etc... in the municipality of Hässleholm in Sweden.
Average in diff. languages (programming), a saddlers apprentice in my youth (leather virtuoso in my spare time nowadays) and a poet...
...I see the possibilities of winget in my work with SCCM/applications (if many flaws can be sorted out)!

**Reply by Romanitho** [reply]
Nice. Welcome ;)

**Comment by patrikcrhp** [comment]
This is a really good project. I have so far been able to deploy various apps using Winget and Intune as an "Required" install. 

However, when I try to make Winget apps available in the Company Portal for the users to install themselves (Users are Standard user), the apps installs very quickly without actually being installed.

Both Company Portal and Intune will state the app has been successfully installed, while if i Run PowerShell > CMD "winget list" on the endpoint device, the app does not show on the list as installed.

The install logs does not show any entries for app install or download, for any the apps installed through Company Portal.

Does this script not currently work with Company Portal as deployment scenario or am I just missing something?

**Reply by Romanitho** [reply]
Hello. For me it works as "available" app. Can you share the way you do it ?

**Reply by patrikcrhp** [reply]
Thanks for the quick response. 

This is my process:
1. I set the ID in winget-install.ps1 to ex. "Lenovo.SystemUpdate" 
2. I do the same in winget-detect.ps1 under  &AppToDetect "Lenovo.SystemUpdate"
3. I convert winget-install.ps1 using IntuneWinAppUtil.exe to .intunewin-file and uploade the file to my test-tenant.
4. Below is the configuration done in Intune during upload:

- Name: Lneovo.SystemUpdate
- Description: Lenovo System Update keeps your drivers and firmware up-to-date.
- Publisher: Lenovo
- Install command: powershell.exe -Executionpolicy Bypass -Command .\winget-install.ps1 -AppIDs Lenovo.SystemUpdate
- uninstall command: powershell.exe -Executionpolicy Bypass -Command .\winget-install.ps1 -AppIDs Lenovo.SystemUpdate -uninstall
- Install behaviour: User
- Requirements: x86, x64
- Minimum Operating System: 1607
- Rules format: User a customer Detection Script > I use winget-detect.ps1
- Assignment "Available"

Shortly after the Application appear in the Company Portal on the Endpoint Device, But when I press Install, it show the install progress for about  3-5 seconds and states installation successful afterwards. However, it does not seem to install anything but regardless Company Portal and Intune states that the application is currently installed correctly, despite the application not existing on the endpoint device at all.

**Reply by Romanitho** [reply]
Hi,

Not sure to understand your step 1. You don't have to change anything inside the winget-install.ps1 as you put your ID in the install command. Weird as well that your detection script "sees" something installed.

**Reply by KnifMelti** [reply]
> * Install behaviour: User


You said in your first post that "../Users are Standard user/.."
Are the installations also installable in user context?
Also, maybe you shouldn't make an intunewin of it but install as a win32 application?

**Reply by patrikcrhp** [reply]
Well it seem it was all related to a mistake of my own. 

I had a another script copying Winget-install.ps1 and Winget-Detect.ps1 to a newly created directory in the source-folder before initiating a conversion of the winget-install.ps1 to .intunewin-file and uploading the file to the tenant.  

My script did not copy the edited content of winget-detection.ps1 and install-winget.ps1 before conversion. And since I have been adding IDs to winget-install.ps1 per app, it may have broken something. 

Even better - I thought I had to add the application ID to the winget-install.ps1 and convert to .intunewin and upload a version per application, but it works with original winget-install as you described. 

It's really promising so far!

**Reply by KnifMelti** [reply]
Just take heed of the SysNative problem in Intune...
https://github.com/Romanitho/Winget-AutoUpdate/discussions/88
...ah, that doesn't apply here?

**Reply by Romanitho** [reply]
https://github.com/Romanitho/WingetIntunePackager

**Comment by nips88** [comment]
Hi,
I have been using this script to install apps on my end user devices from past 2 months and everything was working fine - apps are getting installed, are getting added to WAUWhitelist and creating logs. However, now the script is not adding anything to WAUWhitelist. I haven't made any changes to script or updated with new one. Just added another application to intune with this script and now none of the apps are  getting added to the auto update list. In log there is no entry related to whitelisting. To debug, I added some lines for logging and those 2 lines appear in log, but seems there is something wrong. Manually, script is working just fine.
Any idea? Thanks! 
[winget-install.txt](https://github.com/Romanitho/Winget-Install/files/9131962/winget-install.txt)
[install.log](https://github.com/Romanitho/Winget-Install/files/9131976/install.log)

**Reply by Romanitho** [reply]
Hi,

Have you changed your install command with the "sysnative" one ?

**Reply by nips88** [reply]
Yes, tried but that didn't work. Found that the script is not reading the registry value, and even if it is not reading the reg value, the value of $whitelist variable was "\included_apps.txt" (as if in working directory). Now I have added **$WhiteList = "$env:ProgramData\Winget-AutoUpdate\included_apps.txt"** in function **Add-WAUWhiteList** and it is working fine. 
One more thing, v1.8.0 isn't working at all in my environment. For every app, it says couldn't find app in winget repository, while the same app can be installed using v1.7.4.

**Reply by Romanitho** [reply]
Thanks for your investigations. I'll have a look on that when I have few minutes.

**Comment by Thejuice336** [comment]
I set up a test app in SCCM using the install, uninstall and detection examples provided in the documentation, but when I check the deployment status it says Error 0x87D00327 (-2016410841) Script is not signed. Do I need to sign the detection method script since it doesn't have an execution policy bypass option or set the client settings to bypass?

**Reply by KnifMelti** [reply]
That's your domain kicking in that you can change via GPO, nothing you can do in SCCM:
https://4sysops.com/archives/set-powershell-execution-policy-with-group-policy/
Or, as you mentioned, you must sign the script.

**Reply by Thejuice336** [reply]
I did request a code signing certificate from our internal CA and signed both the winget-install and winget-detect script and still received the same evaluation failed 0x87D00327 (-2016410841) error. The policy on that test machine was set to the default execution policy of restricted. I set the execution policy to AllSigned locally and tested the signed version of the script locally as well and it prompted if I trusted this code signing user and then ran the script after saying yes. Guessing the winget-detection.ps1 script portion needing to be signed is what's causing this error as the install command for winget-install.ps1 should be bypassing this with the -ExecutionPolicy bypass switch in the installation command.
This code signing certificate has a one year duration, so I wouldn't want to resign all of the apps that we use this with unless there is a better way besides setting the policy in the SCCM client settings to bypass.

**Reply by Thejuice336** [reply]
Tried setting the execution policy with local GPO as both RemoteSigned and AllSigned, but still seeing the evaluation failed 0x87D00327 (-2016410841) error under deployment status for that application deployment. Verified with Get-ExecutionPolicy -List that it had RemoteSigned listed next to MachinePolicy and AllSigned when I tested with both.

**Reply by KnifMelti** [reply]
No clue then, works as it should in our domain with the mentioned settings in GPO.

**Reply by Thejuice336** [reply]
So it's working for you without signing the install and detection scripts?

**Reply by KnifMelti** [reply]
Yes, we only have the restricted script settings on servers in our domain, not on clients, via GPO.
So, no signing yet.

**Reply by Thejuice336** [reply]
So clients are set to either bypass or unrestricted?

**Reply by KnifMelti** [reply]
Not at work now, let me check in detail our GPO settings when I am.

**Reply by Thejuice336** [reply]
So, setting it to unrestricted with local GPO and to install for user instead of system seems to have gotten it to work for this Microsoft Remote Desktop app. Maybe there is something wrong with the code signing certificate that I requested\enrolled from our internal CA (Active Directory), which is why SCCM thought it wasn't signed. 

I guess it depends on the app from the store if it can be installed for system, so that it can be used during imaging.

**Reply by Thejuice336** [reply]
So I think I need to deploy my code signing cert with a gpo to the trusted publishers certificate store. Either way too much work for this.

**Reply by KnifMelti** [reply]
Wron gof me.. ..our setting is not in the GPO but in SCCM Client Settings\Computer Agent - PowerShell execution policy = ByPass

**Comment by mmakwana2019** [comment]
Hello Champions, I am trying to deploy Fork.Fork using the winget-install.ps1 through Intune both in system and user context but when the user try's to install the application from company portal, it does not install and says, "the app is not detected." In short i am trying to make the application available in intune and most of the Apps install them in C:\Users\<Username>

**Reply by mmakwana2019** [reply]
To add to above issue, i have created the App using the tool available, but its stuck at 100%

![image](https://github.com/Romanitho/Winget-Install/assets/52775179/5ee836e0-4246-4c70-9836-3287c9dc7697)

**Reply by mmakwana2019** [reply]
After waiting for 2 hours i get this "windows error 0x87D300C9 site:microsoft.com"

![image](https://github.com/Romanitho/Winget-Install/assets/52775179/3d2426c9-3b49-4cdd-a9d5-8fdca1cb9abd)
